doc/guide.md

```diff
@@ -2621,7 +2623,77 @@ the install operation will also deploy and un-deploy as
 necessary. This will only work if the ModuleDescriptor has the
 launchDescriptor property.
 
-### Purge
+### Install jobs and asynchronous operations
+
+For Okapi 4.2.0 and later, the install operation can be asynchronous.
+The asynchronous operation is enabled by URI parameter 'async=true'.
+As for the "synchronous" operation, the dependency check is performed
+first and install/upgrade will return 400 HTTP error upon failure.
+
+Following that, the install operation will create an install "job" on the
+server side and return HTTP status 201 along with a location of the newly
+created install job. The returned JSON content is defined by schema
+([InstallJob.json](../okapi-core/src/main/raml/InstallJob.json)).
+
+This location can then be inspected with HTTP GET for the progress of
+the install operation. The location is same base URI as install, but
+with a slash + the job ID. The install job has properties such as
+
+ * `complete`: boolean which tells whether the job has completed
+ * `id`: job id
+ * `startDate`: start time of job in UTC ISO8601 format.
+ * `endDate`: end time of job in UTC ISO8601 format (only present when complete)
+ * `modules`: enable/disable list of modules along with status.
+
+Each module entry is defined by schema
+([TenantModuleDescriptor.json](../okapi-core/src/main/raml/TenantModuleDescriptor.json)).
+Brief list of properties:
+
+ * `id`: module ID
+ * `from`: old module ID (absent if not upgrading)
+ * `action`: enable/disable/uptodate
+ * `stage`: the current stage of the module
+ * `message`: present upon error (error message)
+
+If `message` property is present an error has occurred and `stage`
+indicates at which stage the error occurred. Stage is one of
+ * `pending`: module is yet to be upgraded/deployed/etc..
+ * `deploy`: module is being deployed
+ * `call`: module is being called via system interface, such as `_tenant`,
+    `_tenantPermissions`, ..
+ * `done`: module is fully upgraded
```

Comment:
Should we mention `undeploy` stage here?

Reply:
Yep.

Should I rename  stage `call` to in `invoke` instead? The reason would be that `invoke` parameter controls whether a module should be invoked/called ..

Reply:
either `call` or `invoke` seems fine to me.